### PR TITLE
Enable WSF/GSF for H2STORE

### PR DESCRIPTION
**Pull request by svenn-t**
Depends on https://github.com/OPM/opm-common/pull/3445

### CODE REVIEW

File: opm/core/props/satfunc/RelpermDiagnostics.cpp [hunk 1]
```diff
@@ -154,9 +154,10 @@ namespace Opm {
         if (family3) {
             const auto& phases = eclState.runspec().phases();
             const bool co2store = eclState.runspec().co2Storage();
-            if ( !(co2store && phases.active(Phase::GAS) && phases.active(Phase::WATER))) {
+            const bool h2store = eclState.runspec().h2Storage();
+            if ( !((co2store || h2store) && phases.active(Phase::GAS) && phases.active(Phase::WATER))) {
                 const std::string msg = "Relative permeability input format: Saturation Family III. \n \
-                                         Only valid for CO2STORE case with GAS and WATER.";
+                                         Only valid for CO2STORE or H2STORE cases with GAS and WATER.";
```
Comment (goncalvesmachadoc) [inline opm/core/props/satfunc/RelpermDiagnostics.cpp:160]: @totto82  Could we allow also for `GASWAT` only (as in the commercial simulator)? Or even to any gas-water systems? Although it is not clear to me what would be the implications of it.

**Comment by totto82** [comment]
jenkins build this please

File: opm/core/props/satfunc/RelpermDiagnostics.cpp (hunk 1 above)
Comment (totto82) [reply opm/core/props/satfunc/RelpermDiagnostics.cpp:160]: I agree. But it should be tested. Do you have a test case where GSF/WSF is used for a gaswater case without co2store or h2store?

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this update_data please

**Comment by totto82** [comment]
jenkins build this opm-tests=1058 please

**Comment by totto82** [comment]
Thanks. This adds basic test for H2STORE and can be merged together with the new reference data.

**Comment by totto82** [comment]
jenkins build this opm-tests=1058 please

**Review by bska — APPROVED** [review]
The new reference solutions have been installed on the CI system.  I'll merge this now to activate these regression tests for H2 storage.

**Pull request merged**